src/coreclr/src/jit/lclvars.cpp

```diff
@@ -3794,7 +3794,8 @@ void Compiler::lvaMarkLclRefs(GenTree* tree, BasicBlock* block, Statement* stmt,
                      allowStructs || genActualType(varDsc->TypeGet()) == genActualType(tree->gtType) ||
                      (tree->gtType == TYP_BYREF && varDsc->TypeGet() == TYP_I_IMPL) ||
                      (tree->gtType == TYP_I_IMPL && varDsc->TypeGet() == TYP_BYREF) || (tree->gtFlags & GTF_VAR_CAST) ||
-                     varTypeIsFloating(varDsc->TypeGet()) && varTypeIsFloating(tree->gtType));
+                     (varTypeIsFloating(varDsc) && varTypeIsFloating(tree)) ||
+                     (varTypeIsStruct(varDsc) == varTypeIsStruct(tree)));
```

Comment:
There are cases where we have a SIMD that we don't normalize when the lclVar is created because the basetype is "Canon". These were getting by before because of the bug in the importer where we were setting `GTF_IND_TGT_ANYWHERE` on the lclVar, which for a lclVar is `GTF_VAR_CAST`, so it passed the above check.
Note that prior to the introduction of SIMD types, struct types would always "match" here.
(Also, I think the `allowStructs` is just wrong and should be eliminated - we can have structs on all targets.)

Reply:
How do we ever see SIMD base types being __Canon? I though these had to be value types.

Reply:
I will say that this is a bit of a mystery to me - it only happens during crossgen, and I've only observed it on a `GT_RET_EXPR` (which is the importer case that was incorrectly setting the flag).